woocommerce/woocommerce-ios
# Fix a memory issue where `IssueRefundViewController` and `RefundConfirmationViewController` are not deallocated after the refund flow

Ref  p91TBi-8fj-p2

### Description


While fixing a similar memory issue for in-person payments/refunds flow, I noticed the `IssueRefundViewController` and `RefundConfirmationViewController` are not deallocated after the refund flow while the coordinator and other classes are gone. After some debugging, it looks like the view controllers are retained by something related to `NotificationCenter` if they are used directly in the second level of callback. I'm still looking for an explanation as in the reference p2.

<img width="1745" alt="Screen Shot 2022-04-18 at 3 11 55 PM" src="https://user-images.githubusercontent.com/1945542/164128495-61de82ee-bc2d-49b4-98b1-717413187146.png">
<img width="1679" alt="Screen Shot 2022-04-18 at 3 20 15 PM" src="https://user-images.githubusercontent.com/1945542/164128505-a7d744af-62ec-4849-912e-715ec45580aa.png">


### Testing instructions


- Go to the orders tab
- Tap on a refundable order
- Tap "Issue Refund"
- Select some items or fees to refund, and tap "Next"
- On the next confirmation screen, drag down to dismiss the refund flow and discard any changes --> in the memory debugger, there should not be any refund objects specific to the flow


---
